src/Common/tests/System/AssertExtensions.cs

```diff
@@ -50,5 +50,26 @@ public static T Throws<T>(string paramName, Func<object> testCode)
 
             return exception;
         }
+
+        public static void Throws(Type netCoreExceptionType, string netCoreParamName, Type netFxExceptionType, string netFxParamName, Action action)
```

Comment:
My PR just got merged with the new `AssertExtensions.Throws<TNetCoreExceptionType, TNetFxExceptionType>` method, so that you can update this PR and use it.